vendor/engines/split_accounts/app/models/split_accounts/split_account.rb

```diff
@@ -36,10 +39,30 @@ def extra_penny_count
     #   subaccounts.where("expires_at IS NOT NULL").order(expires_at: :asc).first
     #
     def earliest_expiring_subaccount
-      subaccounts = splits.map{ |split| split.subaccount }.select(&:expires_at)
+      subaccounts = splits.map(&:subaccount).select(&:expires_at)
```

Comment:
Something always bothers me about the `.select(&:attribute)` construct. Even though I know better, I always initially read it as `map` or `pluck` where the result is the values, when actually it's selecting all the items that have that attribute present.

I'd almost rather see something more like:
`splits.map { |s| s.subaccount.expires_at }.compact.sort.first`

Or even adding `expires_at` and `suspended_at` as delegates on `Split`.

In the expires_at case, the `compact` should be unnecessary as all `Accounts` should have an expiration due to the `validates_presence_of`.